todo-app/src/components/TaskItem.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+
+export default class TaskItem extends React.Component{
+    constructor(props){
+        super(props);
+
+        this.state = {
+            textValue: this.props.task.description
+
+        }
+        this.handleChange = this.handleChange.bind(this)
+        this.handleLocalSubmit = this.handleLocalSubmit.bind(this)
+    }
+    handleChange(change){
+        this.setState({
+            textValue:change.target.value
+        }
+        )
+    }
+    handleLocalSubmit(e){
+        e.preventDefault()
+        this.props.handleSubmit(this.state.textValue)
+    }
+    render(){
+        const {index, task} = this.props
+        return(
+            <form onSubmit={this.handleLocalSubmit}>
+            <input id='completed' 
+            type='checkbox' 
+            defaultChecked={task.done} 
```

Comment:
Instead of using `defaultChecked` and `onClick` here you should turn this into a controller component with `checked` and `onChange`. What do you think?